hazelcast-spring/src/main/java/com/hazelcast/spring/HazelcastConfigBeanDefinitionParser.java

```diff
@@ -1289,6 +1294,59 @@ private void handleDataPersistenceConfig(BeanDefinitionBuilder configBuilder, No
             configBuilder.addPropertyValue("dataPersistenceConfig", dataPersistenceConfigBuilder.getBeanDefinition());
         }
 
+        private void handleTieredStoreConfig(BeanDefinitionBuilder configBuilder, Node node) {
+
+            BeanDefinitionBuilder tieredStoreConfigBuilder = createBeanBuilder(TieredStoreConfig.class);
+            fillAttributeValues(node, tieredStoreConfigBuilder);
+            for (Node child: childElements(node)) {
+                String name = cleanNodeName(child);
+                if ("in-memory-tier".equals(name)) {
+                    handleTSInMemoryTierConfig(tieredStoreConfigBuilder, child);
+                } else if ("disk-tier".equals(name)) {
+                    handleTSDiskTierConfig(tieredStoreConfigBuilder, child);
+                }
+            }
+
+            configBuilder.addPropertyValue("tieredStoreConfig", tieredStoreConfigBuilder.getBeanDefinition());
+        }
+
+        @SuppressWarnings("checkstyle:magicnumber")
+        private void handleTSInMemoryTierConfig(BeanDefinitionBuilder configBuilder, Node node) {
+            BeanDefinitionBuilder tsInMemoryTierConfigBuilder = createBeanBuilder(TSInMemoryTierConfig.class);
+            fillAttributeValues(node, tsInMemoryTierConfigBuilder);
+            for (Node child : childElements(node)) {
+                String name = cleanNodeName(child);
+                if ("capacity".equals(name)) {
+                    tsInMemoryTierConfigBuilder.addPropertyValue("capacity",
+                            getLongValue("capacity", getTextContent(child)) << 20);
+                }
+            }
+
+            configBuilder.addPropertyValue("inMemoryTierConfig", tsInMemoryTierConfigBuilder.getBeanDefinition());
+        }
+
+        @SuppressWarnings("checkstyle:magicnumber")
+        private void handleTSDiskTierConfig(BeanDefinitionBuilder configBuilder, Node node) {
```

Comment:
maybe more explicit name like `handleTieredStoreDiskTierConfig` would be better?